spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/main/java/org/springframework/cloud/function/adapter/gcp/FunctionInvoker.java

```diff
@@ -91,18 +94,38 @@ private <I> Function<Message<I>, Message<byte[]>> lookupFunction() {
 	 */
 	@Override
 	public void service(HttpRequest httpRequest, HttpResponse httpResponse) throws Exception {
+
 		Function<Message<BufferedReader>, Message<byte[]>> function = lookupFunction();
 
 		Message<BufferedReader> message = getInputType() == Void.class || getInputType() == null ? null
 				: MessageBuilder.withPayload(httpRequest.getReader()).copyHeaders(httpRequest.getHeaders()).build();
+
 		Message<byte[]> result = function.apply(message);
 
 		if (result != null) {
+			MessageHeaders headers = result.getHeaders();
 			httpResponse.getWriter().write(new String(result.getPayload(), StandardCharsets.UTF_8));
-			for (Entry<String, Object> header : result.getHeaders().entrySet()) {
-				httpResponse.appendHeader(header.getKey(), header.getValue().toString());
+			for (Entry<String, Object> header : headers.entrySet()) {
+				Object values = header.getValue();
+				if (values instanceof Collection<?>) {
+					StringBuilder sb = new StringBuilder();
+					for (Iterator iterator = ((Collection<?>) values).iterator(); iterator.hasNext();) {
+						sb.append(iterator.next());
+						if (iterator.hasNext()) {
+							sb.append(",");
+						}
+					}
+					httpResponse.appendHeader(header.getKey(), sb.toString());
+				}
+				else {
+					httpResponse.appendHeader(header.getKey(), header.getValue().toString());
+				}
 			}
 			httpRequest.getContentType().ifPresent(contentType -> httpResponse.setContentType(contentType));
+
+			if (headers.containsKey("statusCode")) {
```

Comment:
Extract string literal into a public constant, so users can access it instead of hardcoding the string (which is error-prone).